client/src/http.rs

```diff
@@ -32,4 +32,114 @@ pub trait RequestBuilder {
     /// Sets request's headers
     #[must_use]
     fn headers(self, headers: Headers) -> Self;
+
+    /// Sets request's body in bytes and transforms the builder to its output
+    #[must_use]
+    fn body(self, data: Option<Vec<u8>>) -> Self::Output;
```

Comment:
In general accepting an option in a function argument is a bit of an anti-pattern.

Reply:
The idea is that some `.body()` method should be called anyway, to consume the builder and product its output. I didn't use `Vec::new()` for empty body, because even empty vector allocates some data on stack, which is a **bit** less efficient than using `Option::None` at runtime I think.

https://rust.godbolt.org/z/7nxa4YjGd

Reply:
I think specifying the body and finishing the build can be separated into two different methods

Reply:
Agree... I was focusing on how it is implemented in `http` crate as a reference best-practice example.

Reply:
You probably mean `heap` rather than `stack`. 

The thing is, that you are allocating regardless of which branch is taken. Since Rust prefers `move` assignment, you are not actually improving the code size, just moving the allocation into the body of this function. 

If you really want to elide the heap allocation, you should use `iroha_data_primitives::small::SmallVec` instead. This version does not allocate unless you overflow the original capacity, and can hold up to 24 bytes on the stack.